airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -216,6 +216,7 @@ def ti_update_state(
             kwargs=ti_patch_payload.trigger_kwargs,
         )
         session.add(trigger_row)
+        session.flush()
```

Comment:
Only when i add the session.flush() here, it updates the `trigger` table. Even though we define our session dependency like this: 
```
def _get_session() -> Session:
    with create_session(scoped=False) as session:
        yield session


SessionDep = Annotated[Session, Depends(_get_session)]
```

I am not sure why this is working, tried looking in few docs, but I cannot seem to find a better answer here. @kaxil @ashb any thoughts?

Reply:
Is this only needed to make the unit tests pass?

Reply:
Nope its needed functionally too. Unit tests pass even otherwise, probably cos we call "session.expire_all" after sending the api request

Reply:
Hmmm I feel this "shouldn't" be needed, as create_session calls commit (which internally calls flush) 🤷🏻

Reply:
Yeah!
I think it should work too, it is strange that it doesn't.

Tried reading up some items online, doesn't give me much. I think it is some game fastapi is playing here with its dependency injection